runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -280,6 +289,25 @@ public void onConfigChanged(final ConfigChanged event)
 					panel.setFilterMode(config.subscriptionFilter());
 					updateList();
 					break;
+				case "customWorldCycle":
+					setCustomWorldCycle(event.getNewValue());
+					break;
+			}
+		}
+	}
+
+	private void setCustomWorldCycle(String worldList)
+	{
+		WorldResult worldResult = worldService.getWorlds();
+		worldCycleList.clear();
+		if (worldResult != null || worldList.isEmpty())
+		{
+			if (worldList.matches(NUMBER_COMMA_ONLY_REGEX))
+			{
+				int[] worldCycleListInt = Arrays.stream(config.customWorldCycle().split(",")).mapToInt(Integer::parseInt).filter((x -> (worldResult.findWorld(x) != null))).toArray();
+				//convert valid worlds from string to int
+				Arrays.stream(worldCycleListInt).forEach(x -> worldCycleList.add(worldResult.findWorld(x)));
+				//convert int array to world list
```

Comment:
Don't construct multiple streams to achieve this, join them and put each operation on a new line.

```
Arrays.stream(Text.fromCSV(config.customWorldCycle()))
	.mapToInt(s ->
	{
		try
		{
			return Integer.parseInt(s);
		}
		catch (NumberFormatException e)
		{
			return 0;
		}
	})
	.map(worldResult::findWorld)
	.filter(world -> world != null)
	.forEach(world -> worldCycleList.add(world));
```